### PR TITLE
v2.1 Fix bug, resolve 2 issues, change type to module, add prettier

### LINKED ISSUE #10: Filename parameter
The websocket I need to connect to is ws://127.0.0.1:8888/ws

There's no way to specify '/ws' and so I can't connect. I realize its unconventional, but I can't modify the server application.

### LINKED ISSUE #16: Need to send HEX - no string!
Hey there - I try my best to use this module but I cannot. I guess my pure hex code has to be transformed but in my case this is not possible. With wireshark I see that the sent message adds always the carrier return. 

What can I do?

### PR DESCRIPTION
**Pull request by krzyc**
Change package type to module
Change host+port config to URL (closes #10)
Opt-out from sending new line chars in WS messages (closes #16)
Fix setting variables with JSON path (bug in Companion 3.0 update)
Add prettier config

### CODE REVIEW

File: upgrade.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+export const upgradeScripts = [
+	function v2_1(context, props) {
+		const result = {
+			updatedConfig: null,
+			updatedActions: [],
+			updatedFeedbacks: [],
+		}
+		let config = props.config
+		if (config.url === undefined) {
```
Comment (Julusian) [inline upgrade.js]: `props.config` can be undefined depending in what needs upgrading. You should check that before trying to use it

File: package.json [hunk 2]
```diff
@@ -16,6 +18,7 @@
 		"url": "git+https://github.com/bitfocus/companion-module-generic-websocket.git"
 	},
 	"devDependencies": {
-		"@companion-module/tools": "^1.0.2"
+		"@companion-module/tools": "^1.0.2",
+		"prettier": "^2.8.7"
```
Comment (Julusian) [inline package.json]: Prettier is already installed as part of `@companion-module/tools`, you don't need to install it explicitly. But there is no harm in doing so

File: .prettierrc
Comment (Julusian) [inline .prettierrc:1]: We provide a config you can use as part of `@companion-module/tools`. Check the wiki for how to enable it's use.
You are also welcome to use your own config like this if you prefer

**Review by Julusian — APPROVED** [review]

File: upgrade.js (hunk 1 above)
Comment (krzyc) [reply upgrade.js]: Thanks, fixed.

**Pull request merged**